trieste/space.py

```diff
@@ -91,6 +108,24 @@ def sample(self, num_samples: int) -> tf.Tensor:
 
         return tf.random.shuffle(self._points)[:num_samples, :]
 
+    def __mul__(self, dss: DiscreteSearchSpace) -> DiscreteSearchSpace:
```

Comment:
mypy will probably complain if this isn't
```suggestion
    def __mul__(self, other: DiscreteSearchSpace) -> DiscreteSearchSpace:
```

Reply:
you'll need the same for Box